### PR TITLE
OCP deployment and some fixes

### PR DESCRIPTION
**Pull request by astoycos**
Finishes #224 

and followup fixes from #212

### CODE REVIEW

File: bpfd-operator/config/bpfd-deployment/config.yaml [hunk 1]
```diff
@@ -11,4 +11,5 @@ data:
     key = "/etc/bpfd/certs/bpfd/tls.key"
     client_cert = "/etc/bpfd/certs/bpfd-client/tls.crt"
     client_key = "/etc/bpfd/certs/bpfd-client/tls.key"
-  bpfd.log.level: "info"
\ No newline at end of file
+  bpfd.log.level: "debug"
+
```
Comment (astoycos) [inline bpfd-operator/config/bpfd-deployment/config.yaml:15]: Lol there is a newline :-(

**Comment by astoycos** [comment]
I changed this to a draft just until I add a bug fix reguarding bpfd-operator trying to update a stale bpfProgramConfig object that I forgot about

File: bpfd-operator/pkg/bpfd-agent/bpfd-agent.go [hunk 2]
```diff
@@ -333,28 +279,22 @@ func (r *BpfProgramReconciler) reconcilBpfProgramConfig(ctx context.Context,
 		// Make sure if we're not selected just exit
 		if !isNodeSelected {
 			// Write NodeNodeSelected status
```
Comment (astoycos) [inline bpfd-operator/pkg/bpfd-agent/bpfd-agent.go:277]: ** nit NodeNotSelected

**Review by Billy99 — COMMENTED** [review]
Couple of nits.

General comment: Is `bpfs/bpfd-k8s-api` still used? I thought you deleted in a previous PR but I still see it.

File: bpfd-operator/cmd/bpfd-operator/main.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	bpfdiov1alpha1 "github.com/redhat-et/bpfd/bpfd-operator/api/v1alpha1"
-	"github.com/redhat-et/bpfd/bpfd-operator/pkg/bpfd-operator"
+	bpfdoperator "github.com/redhat-et/bpfd/bpfd-operator/pkg/bpfd-operator"
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/cmd/bpfd-operator/main.go:35]: Good question I'll drop this change

File: bpfd-operator/internal/bpfd-client-helpers.go [hunk 4]
```diff
@@ -29,15 +29,22 @@ import (
...
-	bpfdMapFs             = "/run/bpfd/fs/maps"
-	DefaultConfigPath     = " /etc/bpfd/bpfd.toml"
-	DefaultRootCaPath     = "/etc/bpfd/certs/ca/ca.crt"
-	DefaultClientCertPath = "/etc/bpfd/certs/bpfd-client/tls.crt"
-	DefaultClientKeyPath  = "/etc/bpfd/certs/bpfd-client/tls.key"
+	BpfdDsName             = "bpfd-daemon"
+	BpfdConfigName         = "bpfd-config"
+	BpfdDaemonManifestPath = "./config/bpfd-deployment/daemonset.yaml"
+	bpfdMapFs              = "/run/bpfd/fs/maps"
+	DefaultConfigPath      = " /etc/bpfd/bpfd.toml"
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/internal/bpfd-client-helpers.go:44]: Fixed

File: bpfd-operator/README.md [hunk 5]
```diff
@@ -12,10 +12,14 @@ But first try it out!
 
 ## Getting started
 
+> **Warning**: The bpfProgramConfig name **MUST** match the program's section name.
+
 This operator was build utilizing some great tooling provided by the [operator-sdk library](https://sdk.operatorframework.io/). A great first step in understanding some
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/README.md]: Done

File: bpfd-operator/bundle/manifests/bpfd-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+            "name": "go-counter-example"
+          },
+          "spec": {
+            "attachpoint": {
+              "networkmultiattach": {
+                "interface": "eth0",
+                "priority": 5
+              }
+            },
+            "bytecode": "image://quay.io/bpfd/bytecode:gocounter",
```
[the remark this answers is not in the document]
Comment (astoycos) [reply bpfd-operator/bundle/manifests/bpfd-operator.clusterserviceversion.yaml:23]: I can change if that goes in first

File: bpfd-operator/Makefile [hunk 7]
```diff
@@ -213,15 +105,16 @@ CM_VERIFIER ?= $(LOCALBIN)/cm-verifier
...
 
 KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
 .PHONY: kustomize
 kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
 $(KUSTOMIZE): $(LOCALBIN)
 	test -s $(LOCALBIN)/kustomize || { curl -Ss $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
 
-CM_VERIFIER_BINARY ?= "https://github.com/alenkacz/cert-manager-verifier/releases/download/v0.2.0/cert-manager-verifier_$(CM_VERIFIER_VERSION)_Linux_x86_64.tar.gz"
+## Leave to my own release until https://github.com/alenkacz/cert-manager-verifier/pull/13 merges
+CM_VERIFIER_BINARY ?= https://github.com/astoycos/cert-manager-verifier/releases/download/v$(CM_VERIFIER_VERSION)/cert-manager-verifier_$(CM_VERIFIER_VERSION)_Linux_arm64.tar.gz
```
Comment (anfredette) [inline bpfd-operator/Makefile:117]:
```suggestion
CM_VERIFIER_BINARY ?= https://github.com/astoycos/cert-manager-verifier/releases/download/v$(CM_VERIFIER_VERSION)/cert-manager-verifier_$(CM_VERIFIER_VERSION)_Linux_$(shell arch).tar.gz
```

Comment (astoycos) [reply bpfd-operator/Makefile]: Fixed, good catch thanks!

**Review by anfredette — APPROVED** [review]
lgtm

**Review by Billy99 — APPROVED** [review]
/lgtm

**Pull request merged**